### DISCUSSION TITLE
Outline makes a good blog platform

### DISCUSSION THREAD

**Discussion by routerino** [outline/outline#5067] [Show and tell]
For about a year now I've been drafting my blog in outline, exporting the markdown, running through a markdown translation script, and sticking it in hugo. And it works. But it was messy.

Today I got a bit sick of it and thought: hold on, Outline lets us make external sites now. Can I leverage that?

Turns out you can. And it [looks quite good](https://blog.gurucomputing.com.au/s/blog). I set up a standalone instance and created a collection for sharing, and a collection for drafting. Publishing is as simple as drag-and-dropping the article into the published section. next blog post will be about how it got set up.

## Couple Pain Points

It wasn't all roses. I first tried to reverse proxy my existing wiki and outline did not like that *at all*, most likely due to being a different subdomain. Decided it was better to just make a dedicated blog instance.

Bigger problem I had was getting everything to redirect by default to the blog. Outline will by default redirect to the home page, which didn't work for me. I wanted everything to funnel back to the blog post.

Well some tears and tearing of hair and getting familiar with the chrome debug console later, I figured out which endpoint paths I had to exempt to keep the wiki working while also defaulting redirection to the shared blogpost. (A configuration setting to set the default redirect would be useful here).

Here's what *that* caddy file ended up looking like.

```
@blog {
	host blog.gurucomputing.com.au
}
handle @blog {
	@notBlog {
		not path /static*
		not path /doc*
		not path /s/*
		not path /auth*
		not path /realtime*
		not path /images*
		not path /locales*
		not path /api*
		not path /collaboration*
		not path /collection*
	}
	redir @notBlog /s/blog
	reverse_proxy http://blogoutline
}

#minio
@blogdata-admin {
	host blogdata-admin.gurucomputing.com.au
	import localSubnets
}
handle @blogdata-admin {
	reverse_proxy http://blogoutline_minio:9001
}
@blogdata host blogdata.gurucomputing.com.au
handle @blogdata {
	reverse_proxy http://blogoutline_minio:9000
}
```

But overall it came out really well! Would recommend. Will do a proper write up in the future once it's pressure tested a bit. Also I have to ~turn off those redirections to log in if I get logged out~ (fixed with adding the `/auth*` path to the exceptions).

Oh and analytics of course! I'm in the process of hooking up google analytics (which actually was a headache because it's blocked at my router due to adblock settings), but we'll see how that goes. I was using selfhosted [plausible](https://plausible.io/) before, but that's not possible without injecting some javascript mid flight. maybe a way to add static javascript to every page in the admin settings would be good? Also CSS Styling overrides for shared sites (I'm actually happy with the minimalist look, but I know that people like to make their blogs unique feeling).

@tommoor kudos for making such an incredible knowledgebase platform.

**Comment by Snuupy** [comment]
I'd totally use outline as my blogging platform too - if only there was a native blog integration for outline.
